src/providers/local/exit-pool.provider.ts

```diff
@@ -172,6 +185,21 @@ const provider = (props: Props) => {
     shouldSignRelayer.value ? [relayerApprovalAction.value] : []
   );
 
+  const shouldUseSwapExit = computed(
+    (): boolean =>
+      isSingleAssetExit.value &&
+      isDeep(pool.value) &&
+      isPreMintedBptType(pool.value.poolType)
+  );
+
+  const exitHandlerType = computed((): ExitHandler => {
+    if (shouldUseSwapExit.value) return ExitHandler.Swap;
+    if (isWeightedPool.value && isSingleAssetExit.value)
+      return ExitHandler.ExactOut;
```

Comment:
Why did you change this logic? If the user has clicked the max button we need to do ExactIn so that the full BPT balance is used. Otherwise, you can end up with a dust situation.

Reply:
Right 👍. Changed it back.